ktlint-rule-engine-core/src/main/kotlin/com/pinterest/ktlint/rule/engine/core/api/TokenSets.kt

```diff
@@ -0,0 +1,38 @@
+package com.pinterest.ktlint.rule.engine.core.api
+
+import org.jetbrains.kotlin.com.intellij.psi.tree.TokenSet
+
+public object TokenSets {
+    public val COMMENTS: TokenSet = TokenSet.create(ElementType.BLOCK_COMMENT, ElementType.EOL_COMMENT, ElementType.KDOC)
+    public val EXPRESSIONS: TokenSet =
```

Comment:
> No, I didn't get this from a specific source unfortunately. When I mentioned that there might be bugs, these token sets are one of my biggest worries. Like EXPRESSIONS, I'm afraid it could be missing something that didn't get covered by a test.

This is rather concerning.

> The kotlin compiler has some like [KtTokenSets](https://github.com/JetBrains/kotlin/blob/0fe3576a7c32cbb63f0f6812281115f5daea6f27/compiler/psi/src/org/jetbrains/kotlin/psi/stubs/elements/KtTokenSets.java#L12) but I am not sure if I've seen one with all expressions.

https://kotlinlang.org/docs/reference/grammar.html#expression gives some insights, but it can not be simply match to the tokens either.